src/benchmarking/directory_parser.py

```diff
@@ -0,0 +1,207 @@
+
+    with open(output_csv_path, 'w') as csvfile:
+        writer = csv.writer(csvfile) 
+        writer.writerow(['Trial_Num', 'Experiment', 'Condition', 'Tool', 'File_Path'])
+
+    # parse truth
+    truth_rows = parse_truth_filepaths(truth_path)
+    file = write_to_csv(truth_rows, output_csv_path)
+
+    # parse results
+    result_rows = parse_results_filepaths(result_path)
+    file = write_to_csv(result_rows, output_csv_path)
+    
+    print(f'Metadata saved to:{file}')
+
+
+def parse_truth_filepaths(dir_path: str) -> list:
+    '''
+    DESCRIPTION:
+        Get metadata from truth directory paths and store in list
+        
+    INPUT:
+        Path to all truth files (called "tests" in our case)
+    
+    OUTPUT:
+        A list of lists. Each row contains the metadata for each "truth" entry
+    '''
+
+    rows = []
+
+    for trial in os.listdir(dir_path):
+        if trial.startswith('.'):
+            continue
+        try:
+            trial_num = int(''.join(filter(str.isdigit, trial)))
+        except ValueError as e:
+            trial_num = 0
+            print(f"Directory {trial} does not have an integer, causing error: {e}", file=sys.stderr)
+        finally:
+            trial_path = Path(dir_path) / trial
+            if os.path.isdir(trial_path):
+                for test in os.listdir(trial_path):
+                    test_path = Path(trial_path) / test
+                    if os.path.isdir(test_path) and 'config' not in test:
+                        for subtest in os.listdir(test_path):
+                            subtest_path = Path(test_path) / subtest
+                            if os.path.isfile(subtest_path):
+                                if subtest.endswith('.fa'):
+                                    info = [trial_num, test, subtest.strip('.fa'), 'TRUTH', os.path.abspath(subtest_path)] 
+                                    rows.append(info)
+                 
+    return rows
+
+
+
+def parse_results_filepaths(results_path: str) -> list:
+    '''
+    DESCRIPTION:
+        Get metadata from tools' results directory paths and store in list
+        
+    INPUT:
+        Path to all tools' results files.
+        Expected directory structure for results: 
+            results/<trial num>/<tool name>/<experiment>/<condition>/
+        Example expected structure:
+            results/trial_2/EnrichSeq/num_genomes/200_genomes_500000_reads/...
+        * up to 9 trials
+    
+    OUTPUT:
+        A list of lists. Each row contains the metadata for each tool's results
+    '''
+    result_rows = []
+    for trial in os.listdir(results_path):
+        if trial.startswith('.'):
+            continue
+        try:
+            trial_num = int(''.join(filter(str.isdigit, trial)))
+        except ValueError as e:
+            trial_num = 0
+            print(f"Directory does not have an integer, causing error: {e}", file=sys.stderr)
+        finally:
+            trial_path = Path(results_path) / trial
```

Comment:
Same here - using recursive approach cleans the number for-loops. The stopping criteria here could be the depth